registry/storage/driver/s3-aws/s3.go

```diff
@@ -970,8 +970,16 @@ func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, pre
 	defer done("s3aws.ListObjectsV2Pages(%s)", path)
 	listObjectErr := d.S3.ListObjectsV2PagesWithContext(ctx, listObjectsInput, func(objects *s3.ListObjectsV2Output, lastPage bool) bool {
 
-		*objectCount += *objects.KeyCount
-		walkInfos := make([]walkInfoContainer, 0, *objects.KeyCount)
+		var count int64
+		if objects.KeyCount != nil {
```

Comment:
I think it's worth adding some comment here that certain implementations of the S3 API do not completely implement this.

Reply:
added a meaningful comment. Didn't squash as the original patch is not my work